paper.md

```diff
@@ -38,89 +38,140 @@ affiliations:
 
-*TODO: Nene and Vilém - how the results work together (KM+template)*
 
 # Querying DAISY Data in DSW
 
-*TODO: Marek - describe goals and motivation*
+The other independent approach that we decided to implement uses linking a DAISY project from DSW and then querying data using DAISY API when needed. The overal idea is shown in Figure 1. User of DSW works on its project (with a goal to get a DMP but not limited to it) and in the questionnaire selects using an integration question one of the projects in DAISY. Now, the project from DAISY is linked via its link stored as a reply in DSW project. When user wants to produce a document, e.g., a DMP or a report, current DAISY project data are queried using DAISY REST API and transformed into the desired format. The user gets a document with data from DAISY project appropriately included within the other data originated directly from replies in DSW.
 
-*TODO: Jan - schema how it works*
+![Design of DSW-DAISY Integration](./figures/integration-workflow.png)
+
+With such approach, data in DAISY can be independently changed while selected project in DSW remains the same (as reply in a questionnaire). Creating a document makes a snapshot of both DSW and DAISY projects using the selected document template. First, we needed to investigate how integration question in DSW can be used and how it must be extended to allow users select own projects from DAISY. Then, we had to solve how get the data from DAISY inside a DSW document template that uses [Jinja2 language](https://jinja.palletsprojects.com).
 
 ## Integration Question for DAISY Projects
 
-*TODO: Marek - motivation, describe issues with integration question*
+DSW already supports integration question since v1.6.0 (April 2019). It allows to query a REST API for type hints – as user types reply, it suggests answers provided from the API response, and user can select the reply from integration. The advantage is configurability of the integrations and the stored ID of selected item. On the other hand, it does not allow any form of user authentication towards the external API nor more complex item selection than just regular input dropdown. In Figure 2, you can see how the integration works for selecting affiliation from [Research Organization Registry (ROR)](https://ror.org).
+
+![Integration question for affiliation using ROR.org](./figures/original-integration-question.png)
+
+We tried that it is possible to create such an integration for DAISY projects. However, it has two pitfalls:
```

Comment:
We attempted to use this integration feature as is to query DAISY projects. However, it has two pitfalls:

Reply:
```suggestion
We attempted to use this integration feature as is to query DAISY projects. However, it has two pitfalls:
```